lib/vm.js

```diff
@@ -286,6 +287,72 @@ function runInThisContext(code, options) {
   return createScript(code, options).runInThisContext(options);
 }
 
+function compileFunction(code, params, options) {
+  const defaults = {
+    filename: '',
+    columnOffset: 0,
+    lineOffset: 0,
+    cachedData: undefined,
+    produceCachedData: false,
+    parsingContext: undefined
+  };
+
+  if (typeof code !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('code', 'string', code);
+  }
+  if (params !== undefined) {
+    if (!Array.isArray(params)) {
+      throw new ERR_INVALID_ARG_TYPE('params', 'Array', params);
+    }
+    params.forEach((param, i) => {
+      if (typeof param !== 'string') {
+        throw new ERR_INVALID_ARG_TYPE(`params[${i}]`, 'string', param);
+      }
+    });
+  }
+
+  if (options !== undefined) {
+    for (const key in options) {
```

Comment:
I'd prefer *not* using `for`-`in` loops for checking types... it doesn't scale well when we want to add new options with odd types.

Reply:
Any suggestions on what I should use instead? I think I should with `Object.keys(options).forEach(...)`, but I not a lot of alternatives come to mind.